styx-common/src/main/java/com/spotify/styx/util/Connections.java

```diff
@@ -54,6 +56,7 @@ public static Datastore createDatastore(Config config) {
     return DatastoreOptions.builder()
         .namespace(namespace)
         .projectId(projectId)
+        .retryParams(RetryParams.defaultInstance())
```

Comment:
This seems to be the default behaviour which means by default it retries. Check `ServiceOptions` L217. We have been doing something on top of the default retry.